momepy/preprocessing.py

```diff
@@ -740,3 +745,368 @@ def _get_extrapolated_line(coords, tolerance, point=False):
+    Consolidate close street intersections into a single node, collapsing short edges.
+
+    If rebuild_graph is True, new edges are drawn according to rebuild_edges_method which is one of:
+
+    1. Linear reconstruction:
+        Edges are linearly extended from original endpoints until the new nodes. This method preserves
+        most faithfully the network geometry.
+    2. Spider-web reconstruction:
+        Edges are cropped within a buffer of the new endpoints and linearly extended from there. This
+        method improves upon linear reconstruction by mantaining, when possible, network planarity.
+    3. As-the-crow-flies reconstruction:
+        Edges are ignored and new edges are built as straightlines between new origin and new
+        destination. This method ignores geometry, but efficiently preserves adjacency.
+
+    If rebuild_graph is False, graph is returned with consolidated nodes but without reconstructed
+    edges i.e. graph is intentionally disconnected.
+
+    Graph must be configured so that
+
+    1. All nodes have attributes determining their x and y coordinates;
+    2. All edges have attributes determining their origin, destination, and geometry;
+    3. Graph itself has an attribute determining its crs.
+
+    Parameters
+    ----------
+    graph : Networkx.MultiGraph or Networkx.MultiDiGraph
+
+    tolerance : float
+        distance in network units below which nodes will be consolidated
+
+    rebuild_graph: Boolean
+
+    rebuild_edges_method: string
+        'spider' or 'linear' or 'crow_flies'
+
+    directed: Boolean
+
+    x_col, y_col: string
+        node attribute with the valid coordinate
+
+    edge_from_col, edge_to_col: string
+        edge attribute with the valid origin/destination node id
+
+    Returns
+    ----------
+    Networkx.MultiGraph or Networkx.MultiDiGraph
+
+    """
+
+    # Collect nodes and their data:
+    nodes, nodes_dict = zip(*graph.nodes(data=True))
+    nodes_df = pd.DataFrame(nodes_dict, index=nodes)
+    nodes_gdf = gpd.GeoDataFrame(
+        nodes_df,
+        crs=graph.graph["crs"],
+        geometry=gpd.points_from_xy(nodes_df[x_col], nodes_df[y_col]),
+    )  # improve crs here
+
+    # In case we did not specify directionality, we infer it from the network:
+    if directed is None:
+        directed = True if isinstance(graph, nx.MultiDiGraph) else False
+
+    # Create a graph without the edges above a certain length and clean it from isolated nodes (the unsimplifiable nodes):
+    components_graph = deepcopy(graph)
+    components_graph.remove_edges_from(
+        [
+            edge
+            for edge in graph.edges(keys=True, data=True)
+            if edge[-1]["length"] > tolerance
+        ]
+    )
+    isolated_nodes_list = list(nx.isolates(components_graph))
+    components_graph.remove_nodes_from(isolated_nodes_list)
+
+    # The connected components of this graph are node clusters we must individually simplify. We collect them in a dataframe and
```

Comment:
try to make sure that all lines, including comments, are shorter than 88 characters. I think that `flake8` will complain here, though we don't have it on in CI. (@jGaboardi maybe we should? also `isort`?)

Reply:
Seems to be some good options for [`flake8`](https://github.com/py-actions/flake8) and [`isort`](https://github.com/isort/isort-action) with GHA. Would not hurt at all to give it a try. Also, [`isort`+`black` compatibility with `pre-commit`](https://pycqa.github.io/isort/docs/configuration/black_compatibility.html).